pkg/operator/controller/garden/garden/reconciler_reconcile.go

```diff
@@ -322,6 +322,11 @@ func (r *Reconciler) reconcile(
 			Fn:           c.kubeStateMetrics.Deploy,
 			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying Gardener Metrics Exporter",
+			Fn:           c.gardenerMetricsExporter.Deploy,
+			Dependencies: flow.NewTaskIDs(waitUntilKubeAPIServerIsReady, deployGardenerResourceManager),
```

Comment:
1. We should also wait for the `gardener-apiserver` here after #8309 is merged, right?
1. Why is the KSM deployment task above not depending on `waitUntilKubeAPIServerIsReady` as well?

Reply:
> Why is the KSM deployment task above not depending on waitUntilKubeAPIServerIsReady as well?

Not sure. I will add `waitUntilKubeAPIServerIsReady` as a dependency for KSM.